quinn-proto/src/frame.rs

```diff
@@ -441,10 +443,95 @@ impl ApplicationClose {
     }
 }
 
-// TODO(@divma): for now reusing the struct, good or bad idea?
```

Comment:
made it two structs, there is some duplication but I think the logic overall becomes clearer